### ISSUE TITLE
Erroneous extra intendation after raw string

### ISSUE THREAD

**Issue by louwers** [vim/vim#2019]
After writing a raw string literal with the cindent feature enabled, Vim erroneously indents. 

To reproduce: 
```
vim -N -u NONE 
:set cindent<CR> 
```
Insert the following text (the part before the arrows): 
```
int main() { 
       R"(   <--- proper indentation (6 spaces) 
hello       <--- no indentation, as desired for raw string literals 
world)"; 
             <--- vim erroneously indents again (12 spaces in total) 
```
The problem is that it doesn't recognize the raw string as terminated. `LOOKFOR_UNTERM` gets set in `get_c_indent` in `misc1.c` at line [8661](https://github.com/vim/vim/blob/master/src/misc1.c#L8661).

**Comment by chrisbra** [comment]
can you check, if this [patch](http://256bit.org/~chrisbra/zeugs/0001-cindent-better-support-for-raw-strings.patch) fixes it for you?

**Comment by louwers** [comment]
@chrisbra It does!

**Comment by louwers** [comment]
Here is a test:

```
func Test_raw_string()
  new
  setlocal cindent

  call feedkeys("i" .
        \ "int main() {\<CR>" .
        \ "R\"(\<CR>" .
        \ ")\";\<CR>" .
        \ "statement;\<Esc>", "x")
  assert_equal(getline(line('.')), "\tstatement;")
endfunc
```

**Comment by vim-ml** [comment]
On Do, 24 Aug 2017, Bart wrote:

> Here is a test:
> 
> ```
> func Test_raw_string()
>   new
>   setlocal cindent
>   norm i
> 
>   call feedkeys("i" .
>         \ "int main() {\<CR>" .
>         \ "R\"(\<CR>" .
>         \ ")\";\<CR>" .
>         \ "statement;\<Esc>", "x")
>   assert_equal(getline(line('.')), "\tstatement;")
> endfunc
> ```

Thanks. It modified slightly and added it to the patch. Result is 
attached to this mail. Bram, please include.

Best,
Christian
-- 
Was ist der Unterschied zwischen einem Fuchs und einem Wessi?  Der
Fuchs ist schlau und stellt sich dumm - der Wessi macht es andersrum.

**Comment by vim-ml** [comment]
Christian Brabandt wrote:

> On Do, 24 Aug 2017, Bart wrote:
> 
> > Here is a test:
> > 
> > ```
> > func Test_raw_string()
> >   new
> >   setlocal cindent
> >   norm i
> > 
> >   call feedkeys("i" .
> >         \ "int main() {\<CR>" .
> >         \ "R\"(\<CR>" .
> >         \ ")\";\<CR>" .
> >         \ "statement;\<Esc>", "x")
> >   assert_equal(getline(line('.')), "\tstatement;")
> > endfunc
> > ```
> 
> Thanks. It modified slightly and added it to the patch. Result is 
> attached to this mail. Bram, please include.

I'll include it, thanks.

-- 
"Making it up?  Why should I want to make anything up?  Life's bad enough
as it is without wanting to invent any more of it."
		-- Marvin, the Paranoid Android in Douglas Adams'
		   "The Hitchhiker's Guide to the Galaxy"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by chrisbra** [comment]
fixed by [8.0.0999](https://github.com/vim/vim/releases/tag/v8.0.0999)
